man/foreign.doc

```diff
@@ -3238,10 +3241,11 @@ be used after a call to PL_next_solution() that returns \const{FALSE}
 when the query was created using the \const{PL_Q_PASS_EXCEPTION} flag
 (see PL_open_query()).
 
-Note that an API may only raise an exception when it fails. The
+Note that an API may only raise an exception when it fails; if the API
+call succeeds, the result of \exam{PL_exception(0)} is undefined. The
```

Comment:
I did another edit to this.    "Undefined" typically means it is illegal to call PL_exception(0) in that case.   That is not the case.  It returns 0 (no exception) unless there was already a pending exception before calling the API.   This however should not happen in a well behaved client.     This leads to a bit complicated explanation and is why I simply ignored this case earlier, assuming nobody would do this anyway.     In principle, user code can call `if ( PL_exception(0) ) return FALSE` at any time within the definition of a foreign predicate.